src/Supervisor.cpp

```diff
@@ -1088,4 +1088,93 @@ ZunResult Supervisor::PlayAudio(char *path)
     return ZUN_SUCCESS;
 }
 #pragma optimize("", on)
+
+// u32 g_ControllerData[32];
+DIFFABLE_STATIC_ARRAY(u32, 32, g_ControllerData)
+
+#pragma optimize("s", on)
+u8 *th06::Controller::GetControllerState()
+// this is for rebinding keys
+{
+    // DWORD_PTR cookie;
+    MMRESULT MVar1;
+    HRESULT HVar2;
+    u32 *piVar3;
+    int i;
+    u32 puVar3;
+    joyinfoex_tag *joyinfo_ptr;
+    u32 *puVar4;
+    int local_retryCount;
+    DIJOYSTATE2 local_15c;
+    int local_44;
+    u32 local_40;
+    u32 local_3c;
+    joyinfoex_tag joyinfo;
+
+    memset(&g_ControllerData, 0, sizeof(g_ControllerData));
+    if (g_Supervisor.controller == (LPDIRECTINPUTDEVICE8A)0x0)
+    {
+        // TODO: not tested
+        joyinfo_ptr = &joyinfo;
+        for (i = 13; i != 0; i = i + -1)
+        {
+            joyinfo_ptr->dwSize = 0;
+            joyinfo_ptr = (joyinfoex_tag *)&joyinfo_ptr->dwFlags;
+        }
```

Comment:
This whole pattern is a memset.

```suggestion
        memset(joyinfo, 0, sizeof(JOYINFOEX));
```